### ISSUE TITLE
Different payment partners for different currencies

### ISSUE THREAD

**Issue by Changaco** [liberapay/liberapay.com#1081]
This issue is an offshoot of #833 to discuss @arunisaac's proposal from <https://social.systemreboot.net/notice/388235>:

> Supporting the Indian Rupee (INR) will make Liberapay extremely useful to a lot of Indians. I know that your current payment processor (Mangopay) does not support the INR, and integrating new currencies is a lot of work for you. But, both @[prashere](https://mastodon.social/users/prashere) and I could help you with the code, and could also find other payment processors that support the INR. Would such a setup with multiple payment processors (current Mangopay and some other payment processor) work for you? What will it take? Do let us know.

Having multiple payment processors for the same currency would create an extra layer of complexity (as previously stated in #374), but having different processors for different currencies wouldn't be as complex, so it might be doable.

Obviously it would still be much simpler to have a single payment partner, but Mangopay doesn't seem to have the will and ability to expand worldwide, so if we want Liberapay to support all major currencies then we need more partners.

@arunisaac What you have to understand is that Mangopay isn't really a payment processor (even though we often call it that for simplicity's sake). Mangopay is an "e-money emitter", they hold our users' funds in "e-wallets" (similar to bank accounts but more limited), and they integrate with various payment processors and banks to support deposits into and withdrawals from those wallets. They also handle verifying the identity of users, as well as other regulatory requirements. Mangopay is like PayPal, except users don't have a single Mangopay account that they can access directly.

The point is that there are only a few potential payment platforms that we can partner with (see #374), because we don't have our own E-Money Emitter license so we can't hold our users' funds in our own bank accounts.

So the first question is: are there e-wallet providers like Mangopay in India?

**Comment by arunisaac** [comment]
I've been searching a lot online. There are many nascent startups providing e-wallets in India. I don't know how trustworthy are reliable such companies are. But, among the larger ones are [Amazon Pay](https://pay.amazon.com/us), [PayTM](https://paytm.com/) and [PhonePe](https://www.phonepe.com/). These are all consumer facing e-wallets where users have accounts, can load money, and spend it on various sites. Are these the kind of e-wallets we want? Or, are "e-money emitters" somehow different?

If Amazon Pay is suitable for our purposes, I think we should go with them. Their technical documentation does claim to support automatic recurring payments which might be useful for us.

**Comment by Changaco** [comment]
No, we don't want consumer-facing e-wallets, we want an e-wallet provider that is mostly invisible (our users should barely know that they're using it). This allows us to have more control and independence.

In other words the provider's API has to allow us to create and manage e-wallets on behalf of users, so that we (almost) never have to redirect them away from our website and towards the provider's website.

**Comment by Changaco** [comment]
It looks like "[payments banks](https://en.wikipedia.org/wiki/Payments_bank)" are similar to "e-money emitters" in what they can do, but they're meant to service consumers directly and they seem to focus on that.

Getting our own banking license isn't doable (see [Banking regulation in India: overview](https://uk.practicallaw.thomsonreuters.com/w-007-9424)).

**Comment by arunisaac** [comment]
Will [SafexPay](https://safexpay.com/) do? They are based in India and claim to provide white label e-wallet solutions. Among other things, they provide "Closed System Wallet" solutions. Closed System Wallet is defined as

> Closed wallets are wallets which are issued by a company to a consumer for buying goods and services exclusively from that company.

See [Prepaid Payment Instruments in India](https://en.wikipedia.org/wiki/Prepaid_Payment_Instruments_in_India). This does seem like what we want.

**Comment by Changaco** [comment]
You beat me to the punch, I was going to post this:

> What we want can be called "white label e-wallets", and searching the web for that returns quite a few results, but most of them only contain very basic information and require contacting the provider to get more. Examples:
>
> https://traxx.asia/products-and-services/e-wallet/
> https://safexpay.com/white-label-solutions.html
> http://www.asiapaytech.com/white-label-electronic-wallet-platform.html
> https://www.digiwallet.online/ (Europe)

**Comment by Changaco** [comment]
@arunisaac The best way to determine if a provider suits our needs is to look at their API documentation. It's not a good sign when they don't publish that, but you can try to contact them anyway.

**Comment by arunisaac** [comment]
Yes, SafexPay hasn't published their API documentation. But, perhaps, they share the API documentation only with their clients. I will contact them and get back to you.

**Comment by arunisaac** [comment]
I couldn't get AsiaPay to respond to my emails either. I don't know what I'm doing wrong. However, the Safexpay website has now been improved allowing people to directly sign up for an account. Maybe, that could be of use to us.

But, considering Liberapay's current troubles with MangoPay and plans for a Liberapay 2.0 #1171, I guess this issue can be suspended for now.

**Comment by arunisaac** [comment]
Thank you! :-)
